### PR TITLE
Get python code to generate tikz figure

### PR DESCRIPTION
**Pull request by 11michalis11**
Create a function that generates the necessary latex code to create a tikz picture of the Markov chains of the ambulance problem. This function is meant to make the creation of Markov chains relatively straight forward.

### CODE REVIEW

File: src/ambulance_game/markov/__init__.py [hunk 1]
```diff
@@ -15,3 +15,8 @@
     mean_waiting_time_formula,
     get_mean_waiting_time_markov,
 )
+
+from .additional import (
+    convert_networkxx_figure_to_tikz,
```
Comment (drvinceknight) [inline src/ambulance_game/markov/__init__.py:20]:
```suggestion
    convert_networkx_figure_to_tikz,
```

?

File: src/ambulance_game/markov/additional.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+import numpy as np
+import tikzplotlib
+
+from .markov import (
+    build_states,
+    visualise_ambulance_markov_chain,
+)
+
+
+def convert_networkxx_figure_to_tikz(
```
Comment (drvinceknight) [inline src/ambulance_game/markov/additional.py:11]:
```suggestion
def convert_networkx_figure_to_tikz(
```

File: src/ambulance_game/markov/additional.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    """TODO: Build a string of latex code that generates the tikz picture of the networkxx model as constructed by the networkxx library.
+    """
+
+    visualise_ambulance_markov_chain(
+        num_of_servers=num_of_servers,
+        threshold=threshold,
+        system_capacity=system_capacity,
+        parking_capacity=parking_capacity,
+    )
+    tikzplotlib.save("example.tex")
```
Comment (drvinceknight) [inline src/ambulance_game/markov/additional.py:23]: What is this for?

File: src/ambulance_game/markov/additional.py [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    Returns
+    -------
+    string
+        A string containing the full latex code to build a tikz figure of the Markov chain
+    """
+    tikz_code = (
+        "\\begin{figure}[h]"
+        + "\n"
+        + "\\centering"
+        + "\n"
```
Comment (drvinceknight) [inline src/ambulance_game/markov/additional.py:52]: Don't have the figure and centering boiler plate.

File: src/ambulance_game/markov/additional.py [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+        + " servers}"
+        + "\n"
+        + "\\label{Exmple_model-"
+        + str(num_of_servers)
+        + str(threshold)
+        + str(system_capacity)
+        + str(parking_capacity)
+        + "}"
+        + "\n"
+        + "\\end{figure}"
```
Comment (drvinceknight) [inline src/ambulance_game/markov/additional.py:219]:
```suggestion

```

File: src/ambulance_game/markov/markov.py [hunk 6]
```diff
@@ -733,7 +734,7 @@ def mean_waiting_time_formula(
         if patient_type == "ambulance":
             mean_waiting_time = np.sum(
                 [
-                    max(min(state[1] + 1, threshold) - num_of_servers, 0)
+                    (min(state[1] + 1, threshold) - num_of_servers)
```
Comment (drvinceknight) [inline src/ambulance_game/markov/markov.py:737]: Why this change?

File: tests/test_markov_model.py [hunk 7]
```diff
@@ -612,3 +614,31 @@ def test_get_mean_waiting_time_from_closed_form_markov():
     assert round(mean_waiting_time, number_of_digits_to_round) == round(
         1.1051493390764142, number_of_digits_to_round
     )
+
+
+def test_generate_code_for_tikz_figure():
```
Comment (drvinceknight) [inline tests/test_markov_model.py:619]: This test looks good, nicer to have them as four separate functions though, so that when the tests fail you might get more immediate information.

**Review by drvinceknight — CHANGES_REQUESTED** [review]
Looking good @11michalis11 :+1:

File: src/ambulance_game/markov/additional.py (hunk 3 above)
Comment (11michalis11) [reply src/ambulance_game/markov/additional.py:23]: This is an extra function that I may or may not look into more. This one generates latex code for a tikz picture of the networkx graph that is built by python. I realise it's not particularly useful and I may never use it, but thought of maybe having it in anyway. Do you think I should just leave it out?

File: src/ambulance_game/markov/markov.py (hunk 6 above)
Comment (11michalis11) [reply src/ambulance_game/markov/markov.py:737]: This is an earlier commit from the other branch that I cherry picked to fix the tests that where failing. It's a completely different thing.

File: tests/test_markov_model.py (hunk 7 above)
Comment (11michalis11) [reply tests/test_markov_model.py:619]: Sure thing. I will separate them now.

File: src/ambulance_game/markov/additional.py (hunk 3 above)
Comment (drvinceknight) [reply src/ambulance_game/markov/additional.py:23]: Let's remove it from the code for now: paste it in to a GitHub issue and come back to it when/if you want/need to. Always good to keep the code base clean.

File: src/ambulance_game/markov/markov.py (hunk 6 above)
Comment (drvinceknight) [reply src/ambulance_game/markov/markov.py:737]: Ok can you remind me why the change anyway? (I'm sure it's correct just trying to remember)

**Review by drvinceknight — APPROVED** [review]

File: src/ambulance_game/markov/additional.py (hunk 3 above)
Comment (11michalis11) [reply src/ambulance_game/markov/additional.py:23]: Sure thing. Will do

File: src/ambulance_game/markov/markov.py (hunk 6 above)
Comment (11michalis11) [reply src/ambulance_game/markov/markov.py:737]: Sure. That is where I calculate the mean waiting time formula, and I used to do that by having a minimisation function inside a maximisation function but in one of our meetings you told me that I could just leave out the max( ) part and simply change the limits that I sum over. And this is what's happening here.

Comment (drvinceknight) [reply src/ambulance_game/markov/markov.py:737]: Ah yes I remember. Thanks :) :+1:

**Comment by drvinceknight** [comment]
This should be merged now I reckon @11michalis11 :+1:
